File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/products/dto/ProductImageResponse.java
package com.portfolio.auctionmarket.domain.products.dto;

import com.portfolio.auctionmarket.domain.products.entity.ProductImage;
import jakarta.persistence.Column;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductImageResponse {
    private Long imageId;
    private Long productId;
    private String imageUrl;
    private Integer imageOrder;


    public static ProductImageResponse from(ProductImage entity) {

        return ProductImageResponse.builder()
                .imageId(entity.getImageId())
                .productId(entity.getProduct().getProductId())
                .imageUrl(entity.getImageUrl())
                .imageOrder(entity.getImageOrder())
                .build();
    }
}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/products/dto/ProductImageResponse.kt
package org.example.auctioncommon.domain.products.dto

import org.example.auctioncommon.domain.product.entity.ProductImage

data class ProductImageResponse(
    val imageId: Long,
    val productId: Long,
    val imageUrl: String,
    val imageOrder: Int,
)
fun ProductImage.toProductImageDto() : ProductImageResponse{
    return ProductImageResponse(
        imageId = this.imageId ?: 0L,
        productId = this.product?.productId ?: 0L,
        imageUrl = this.imageUrl,
        imageOrder = this.imageOrder
    )
}